File: Year-2023/LeetCode-February-Challenge/AddtoArrayFormofInteger.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        for(int i = num.size() - 1; i >= 0; --i){
            num[i] += k;
            k = num[i] / 10;
            num[i] %= 10;
        }
        while(k > 0){
            num.insert(num.begin(), k % 10);
            k /= 10;
        }
        return num;
    }
};
int main(){
    vector<int> num = {2,7,4}; int k = 181;
    Solution *s = new Solution();
    vector<int> res = s->addToArrayForm(num, k);
    for(auto x: res) cout<<x<<" ";
    return 0;
}

File: Year-2023/LeetCode-February-Challenge/AddtoArrayFormofInteger.java
import java.util.*;
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> res = new LinkedList<>();
        for(int i = num.length - 1; i >= 0; --i){
            num[i] += k;
            k = num[i] / 10;
            res.add(0, num[i] % 10);
        }
        while(k > 0){
            res.add(0, k % 10);
            k /= 10;
        }
        return res;
    }
}
public class AddtoArrayFormofInteger{
    public static void main(String[] args) {
    }
}